### DISCUSSION TITLE
[Q&A] How to install the GLPI Agent on SLES SP5?

### DISCUSSION THREAD

**Discussion by AndiMb** [glpi-project/glpi-agent#450] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Is there a way to install the GLPI Agent on SLES SP5. It seems the install script for Linux is not working. I'm getting the following output, when running the Perl installer:

Installing glpi-agent v1.5...
Installing EPEL 12 repository...
sh: dnf: command not found
Can't install EPEL 12 repository:

Regards,
Andreas

**Comment by g-bougard** [comment]
Hi @AndiMb 

I really don't know if the rpm package can work on SLES SP5.

Can you share the output when using also `--verbose` option ?

Have you tried to install directly the rpm, eventually resolving dependencies ? If you find all the process, I can add it to the perl installer script.

Finally, you can try also the snap version or the AppImage package.

**Reply by AndiMb** [reply]
The output with `--verbose` is

```
Found distro: SuSE
Running on linux distro: SUSE Linux Enterprise Server 12 (x86_64) : SuSE : 12...
Installing glpi-agent v1.5...
Trying to install glpi-agent v1.5-1 on SUSE Linux Enterprise Server 12 (x86_64) release (SuSE:12)...
Extracting glpi-agent-1.5-1.noarch.rpm ...
Installing EPEL 12 repository...
Running: dnf -y install epel-release
Can't exec "dnf": No such file or directory at glpi-agent-1.5-linux-installer.pl line 729.
Running: dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-12.noarch.rpm
Can't exec "dnf": No such file or directory at glpi-agent-1.5-linux-installer.pl line 652.
Failed to run dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-12.noarch.rpm: No such file or directory
```
I wasn't able to resolve the dependencies, since there is no perl repo for SLES 15 SP5 as used for OpenSUSE https://download.opensuse.org/repositories/devel:/languages:/perl/

Perhaps I have to try the snap version.

**Reply by g-bougard** [reply]
Okay, so first I see the installer doesn't detect the `SuSE` distro flavour.

Can you confirm we should use `zypper` to install rpm of that system ?

If yes, can you share the output of:
 * `zypper -n repos devel_languages_perl` which effectively check the repos is available,
 * `zypper -n install -y --allow-unsigned-rpm glpi-agent-1.5-1.noarch.rpm` (after downloading the rpm from github) to see how zypper tries to resolv dependencies

**Reply by AndiMb** [reply]
Can you confirm we should use zypper to install rpm of that system ? -> Yes, `zypper` is the tool to install rpms.

```
sles-test:~ # zypper -n repos devel_languages_perl
Repository 'devel_languages_perl' not found by its alias, number, or URI.
Use 'zypper repos' to get the list of defined repositories.
```

```
sles-test:~ # zypper -n install -y glpi-agent-1.5-1.noarch.rpm
Refreshing service 'SUSE_Linux_Enterprise_Server_x86_64'.
Loading repository data...
Reading installed packages...
Resolving package dependencies...

Problem: nothing provides perl(Cpanel::JSON::XS) needed by glpi-agent-1.5-1.noarch
 Solution 1: do not install glpi-agent-1.5-1.noarch
 Solution 2: break glpi-agent-1.5-1.noarch by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/c] (c): c
```

**Reply by g-bougard** [reply]
So what gives `zypper repos` ? Maybe we can identify one to replace 'devel_languages_perl' in that case.

**Reply by hugotrsd** [reply]
The missing Perl dependency can be fixed by adding the repository from https://download.opensuse.org/repositories/devel:/languages:/perl/SLE_15_SP5/.

The RPM seems to work fine on SUSE after that.
